handlers.c

```diff
@@ -432,11 +456,18 @@ bool handler__list(globals_t *vars, char **argv, unsigned argc)
     matches_and_old_values_swath *reading_swath_index = vars->matches->swaths;
     size_t reading_iterator = 0;
 
+    if (ioctl(STDOUT_FILENO, TIOCGWINSZ, &w) == -1) {
+        show_warn("handler__list(): couldn't get terminal size.\n");
+        pager = get_pager(stdout);
+    } else {
+        pager = w.ws_row >= vars->num_matches ? stdout : get_pager(stdout);
```

Comment:
Here you want `max_to_print`.
The warning fires in backend mode as well, but it doesn't really matter.

This whole block could also use an explanatory comment.

Reply:
We want `vars->num_matches`, because `max_to_print` is likely always going to be bigger than the terminal line number. `max_to_print` also doesn't give the actual amount that would be outputted. I added a comment, though.

In backend mode we probably aren't attached to a terminal. I figured it was OK to leave it since we check for backend mode in `get_pager()`.